contrib/envconfig/client_config_load.go

```diff
@@ -0,0 +1,412 @@
+
+	// If file is enabled, load it and find just the profile
+	if !options.DisableFile {
+		// Load
+		conf, err := LoadClientConfig(LoadClientConfigOptions{
+			ConfigFilePath:   options.ConfigFilePath,
+			ConfigFileData:   options.ConfigFileData,
+			ConfigFileStrict: options.ConfigFileStrict,
+			EnvLookup:        options.EnvLookup,
+		})
+		if err != nil {
+			return ClientConfigProfile{}, err
+		}
+		// Find profile or fail. The one situation where we don't fail here is if the profile is not user-set
+		profile := options.ConfigFileProfile
+		profileUnset := false
+		if profile == "" {
+			env := options.EnvLookup
+			if env == nil {
+				env = EnvLookupOS
+			}
+			// Unlike env vars for the config values, empty and unset env var
+			// for config file path are both treated as unset
+			profile, _ = env.LookupEnv("TEMPORAL_PROFILE")
+		}
+		if profile == "" {
+			profile = DefaultConfigFileProfile
+			profileUnset = true
+		}
+		if profPtr := conf.Profiles[profile]; profPtr != nil {
+			prof = *profPtr
+		} else if !profileUnset {
+			return ClientConfigProfile{}, fmt.Errorf("unable to find profile %v in config data", profile)
+		}
+	}
+
+	// If env is enabled, apply it
+	if !options.DisableEnv {
+		if err := prof.ApplyEnvVars(options.EnvLookup); err != nil {
+			return ClientConfigProfile{}, fmt.Errorf("unable to apply env vars: %w", err)
+		}
+	}
+
+	return prof, nil
+}
+
+// DefaultConfigFileProfile is the default profile used.
+const DefaultConfigFileProfile = "default"
+
+// DefaultConfigFilePath is the default config file path used. It is [os.UserConfigDir]/temporal/temporal.toml.
+//
+// WARNING: Environment configuration is currently experimental.
+func DefaultConfigFilePath() (string, error) {
+	userDir, err := os.UserConfigDir()
+	if err != nil {
+		return "", fmt.Errorf("failed getting user config dir: %w", err)
+	}
+	return filepath.Join(userDir, "temporalio", "temporal.toml"), nil
+}
+
+// EnvLookup abstracts environment variable lookup for [ClientConfigProfile.ApplyEnvVars]. [EnvLookupOS] is the common
+// implementation.
+type EnvLookup interface {
+	// Environ gets all environment variables in the same manner as [os.Environ].
+	Environ() []string
+	// Getenv gets a single environment variable in the same manner as [os.LookupEnv].
+	LookupEnv(string) (string, bool)
+}
+
+type envLookupOS struct{}
+
+// EnvLookupOS implements [EnvLookup] for [os].
+var EnvLookupOS EnvLookup = envLookupOS{}
+
+func (envLookupOS) Environ() []string                   { return os.Environ() }
+func (envLookupOS) LookupEnv(key string) (string, bool) { return os.LookupEnv(key) }
+
+// ApplyEnvVars overwrites any values in the profile with environment variables if the environment variables are set and
+// non-empty. If env lookup is nil, defaults to [EnvLookupOS]
+func (c *ClientConfigProfile) ApplyEnvVars(env EnvLookup) error {
+	if env == nil {
+		env = EnvLookupOS
+	}
+	if s, ok := env.LookupEnv("TEMPORAL_ADDRESS"); ok {
```

Comment:
I think it probably makes sense to make these all public constants. It helps with both documentation and, since people can technically implement their own lookup, they might want to programmatically use them.

Reply:
Why not? They could be doing it as part of some infra-as-code stuff

Reply:
Because the constants are part of the spec and I am not sure it has enough value making them part of the public API in every SDK (nor do I think that with the other environment variable names such as `TEMPORAL_DEBUG`).